### PR TITLE
Document issues with kubectl apply

### PR DESCRIPTION
**Pull request by meyskens**
Documenting https://github.com/jetstack/cert-manager/issues/3121

### CODE REVIEW

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 1]
```diff
@@ -5,5 +5,26 @@ weight: 840
...
+## Issue with older versions of `kubectl`
+`kubectl` versions lower than `v1.19.0-rc.1` have issues updating the `v0.16` CRD files, this is due to [a bug when handling deeply nested CRDs](https://github.com/kubernetes/kubernetes/issues/91615).
+This patch is being backported into older versions of `kubectl` as new patch releases soon, at time of writing this was not the case.
+This bug will make `kubectl apply -f [...]` hang. 
+
+This bug only happens if you try to re-apply the `v0.16` CRDs, initial upgrade will not cause any issues. If you have this issue please use the `v1.19.0-rc.2` version of `kubectl` to apply this.
+*Note: cert-manager does not recommend to use this version to operate older Kubernetes clusters apart from applying the CRD updates.*
+```console
+# On Linux
+$ curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.19.0-rc.2/bin/linux/amd64/kubectl
```
Comment (JoshVanL) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:17]: Adding sha checks is nice to have, not a blocker though

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 2]
```diff
@@ -5,5 +5,26 @@ weight: 840
...
+$ chmod +x ./kubectl
+$ ./kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.crds.yaml
+# If you use the static manifest install
+$ ./kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.yaml
+```
+
+### Helm
+Helm users who use `installCRDs=true` should **not** upgrade to `v0.16` until a Helm version built upon Kubernetes `v1.19` is releases.
+Current testing points out that this issue is only helm `v3.1+` which is affected but we do not advice right now to use `installCRDs=true` in `v0.16`.
```
Comment (JoshVanL) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:28]: This issue only affects `v3.1+` however, we don't currently advice using `installCRDs=true`....

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 3]
```diff
@@ -5,5 +5,26 @@ weight: 840
 type: "docs"
 ---
 
-When upgrading from `v0.15` to `v0.16`, no special upgrade steps are required 🎉.
+## Issue with older versions of `kubectl`
+`kubectl` versions lower than `v1.19.0-rc.1` have issues updating the `v0.16` CRD files, this is due to [a bug when handling deeply nested CRDs](https://github.com/kubernetes/kubernetes/issues/91615).
+This patch is being backported into older versions of `kubectl` as new patch releases soon, at time of writing this was not the case.
+This bug will make `kubectl apply -f [...]` hang. 
+
+This bug only happens if you try to re-apply the `v0.16` CRDs, initial upgrade will not cause any issues. If you have this issue please use the `v1.19.0-rc.2` version of `kubectl` to apply this.
```
Comment (JoshVanL) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:13]: This bug only happens during a re-apply of the `v0.16` CRDs. Initial upgrade does not cause issues.

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 4]
```diff
@@ -5,5 +5,26 @@ weight: 840
...
+*Note: cert-manager does not recommend to use this version to operate older Kubernetes clusters apart from applying the CRD updates.*
```
Comment (JoshVanL) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:14]: To use -> using

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 5]
```diff
@@ -5,5 +5,26 @@ weight: 840
 type: "docs"
 ---
 
-When upgrading from `v0.15` to `v0.16`, no special upgrade steps are required 🎉.
+## Issue with older versions of `kubectl`
+`kubectl` versions lower than `v1.19.0-rc.1` have issues updating the `v0.16` CRD files, this is due to [a bug when handling deeply nested CRDs](https://github.com/kubernetes/kubernetes/issues/91615).
```
Comment (JoshVanL) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:9]: files, due to a

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 6]
```diff
@@ -5,5 +5,26 @@ weight: 840
...
+This patch is being backported into older versions of `kubectl` as new patch releases soon, at time of writing this was not the case.
```
Comment (JoshVanL) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:10]: Soon, however this is not the case at the time of writing.

File: .spelling [hunk 7]
```diff
@@ -156,3 +156,5 @@ Uncomment
 loadbalancer
 propagations
 CertificateRequest
+backported
+v0.16
```
Comment (JoshVanL) [inline .spelling:160]: Newline

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 8]
```diff
@@ -5,5 +5,26 @@ weight: 840
...
+# On macOS
+$ curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.19.0-rc.2/bin/linux/amd64/kubectl
+$ chmod +x ./kubectl
+$ ./kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.crds.yaml
+# If you use the static manifest install
+$ ./kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.yaml
+```
+
+### Helm
+Helm users who use `installCRDs=true` should **not** upgrade to `v0.16` until a Helm version built upon Kubernetes `v1.19` is releases.
```
Comment (urmet) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:27]: typo - "is releases"

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md [hunk 9]
```diff
@@ -5,5 +5,26 @@ weight: 840
...
+$ curl -LO https://storage.googleapis.com/kubernetes-release/release/v1.19.0-rc.2/bin/linux/amd64/kubectl
+$ chmod +x ./kubectl
+$ ./kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.crds.yaml
+# If you use the static manifest install
+$ ./kubectl apply -f https://github.com/jetstack/cert-manager/releases/download/v0.16.0/cert-manager.yaml
+```
+
+### Helm
+Helm users who use `installCRDs=true` should **not** upgrade to `v0.16` until a Helm version built upon Kubernetes `v1.19` is released.
+This issue only affects `v3.1+` however, we don't currently advice using installCRDs=true
```
Comment (JoshVanL) [inline content/en/docs/installation/upgrading/upgrading-0.15-0.16.md:28]: `installCRDs=true`.

File: content/en/docs/installation/kubernetes.md [hunk 10]
```diff
@@ -32,6 +32,10 @@ guides](../../configuration/).
 All resources (the `CustomResourceDefinitions`, cert-manager, namespace, and the webhook component)
 are included in a single YAML manifest file:
 
+> **Note**: If you're using a `kubectl` version below `v1.19.0-rc.1` you will might have issues updating the CRDs.
```
Comment (JoshVanL) [inline content/en/docs/installation/kubernetes.md:35]: you will have issues

File: content/en/docs/installation/upgrading/upgrading-0.15-0.16.md (hunk 9 above)
Comment (urmet) [reply content/en/docs/installation/upgrading/upgrading-0.15-0.16.md]: A bit late, but that should be advise instead of advice

**Pull request merged**